### PR TITLE
fix: unset hash fields using AccountSet

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

Fix [issue#2150](https://github.com/XRPLF/xrpl.js/issues/2150).

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

User can unset a hash field using an empty string with AccountSet similarly to other fields.

### CODE REVIEW

File: packages/ripple-binary-codec/test/hash.test.js [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+  test('can be unset', function () {
+    const h1 = Hash128.from('')
+    expect(h1.toJSON()).toBe('')
+  })
+  test('can be compared against another', function () {
+    const h1 = Hash128.from('100000000000000000000000000000000')
+    const h2 = Hash128.from('200000000000000000000000000000000')
+    const h3 = Hash128.from('000000000000000000000000000000003')
+    expect(h1.lt(h2)).toBe(true)
+    expect(h3.lt(h2)).toBe(true)
```
Comment (mvadari) [inline packages/ripple-binary-codec/test/hash.test.js:18]: Can this also test `.gt`?

**Review by jonathanlei — COMMENTED** [review]
LGTM

File: packages/ripple-binary-codec/src/types/hash-128.ts [hunk 2]
```diff
@@ -9,8 +9,25 @@ class Hash128 extends Hash {
   static readonly ZERO_128: Hash128 = new Hash128(Buffer.alloc(Hash128.width))
 
   constructor(bytes: Buffer) {
+    if (bytes && bytes.byteLength === 0) {
```
Comment (ckniffen) [inline packages/ripple-binary-codec/src/types/hash-128.ts:12]: I think this should happen in Hash similar to how it is done in xrpl-py.

Comment (pdp2121) [reply packages/ripple-binary-codec/src/types/hash-128.ts:12]: Added a PR for xrpl-py:
https://github.com/XRPLF/xrpl-py/pull/508

**Review by ckniffen — APPROVED** [review]

**Pull request merged**